wildmatch.go

```diff
@@ -11,6 +12,13 @@ import (
 type opt func(w *Wildmatch)
 
 var (
+	// Basename allows the receiving Wildmatch to match paths where the
+	// pattern matches only the basename of the path. If the given pattern
```

Comment:
I think the second sentence is a bit confusing in this context, as it doesn't explicitly mention the option itself. I think it would be a little clearer if it said something like `If Basename is false, the entire path will be matched`, and maybe in a separate sentence mention that `Basename` is only set to true if the pattern contains no directory separators.